app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1219,57 +1219,43 @@ export default Controller.extend(bufferedProperty("model"), {
       return Promise.resolve();
     }
     this.model.set("bookmarking", true);
-    const bookmark = !this.model.bookmarked;
-    let posts = this.model.postStream.posts;
+    const alreadyBookmarkedPosts = this.model.bookmarkedPosts;
 
     return this.model.firstPost().then((firstPost) => {
+      const bookmarkPost = async (post) => {
+        const opts = await this._togglePostBookmark(post);
+        this.model.set("bookmarking", false);
+        if (opts && opts.closedWithoutSaving) {
```

Comment:
what would make it possible to not have opts here ?

Reply:
Nothing in this case. I've removed it.